src/Workspaces/MSBuildTest/NewlyCreatedProjectsFromDotNetNew.cs

```diff
@@ -0,0 +1,246 @@
+            var foundDivider = false;
+
+            foreach (var line in lines)
+            {
+                if (!foundDivider)
+                {
+                    if (line.StartsWith("----"))
+                    {
+                        foundDivider = true;
+                    }
+                    continue;
+                }
+
+                var columns = line.Split(new[] { "  " }, StringSplitOptions.RemoveEmptyEntries)
+                    .Select(c => c.Trim())
+                    .ToArray();
+                var templateShortName = columns[1].Split(',').First();
+
+                if (ExcludeMauiTemplates && templateShortName.StartsWith("maui"))
+                    continue;
+
+                templateNames.Add(templateShortName);
+            }
+
+            Assert.True(foundDivider);
+
+            return templateNames;
+        }
+
+        private async Task AssertTemplateProjectLoadsCleanlyAsync(string templateName, string languageName, string[]? ignoredDiagnostics = null)
+        {
+            if (ignoredDiagnostics?.Length > 0)
+            {
+                TestOutputHelper.WriteLine($"Ignoring compiler diagnostics: \"{string.Join("\", \"", ignoredDiagnostics)}\"");
+            }
+
+            var projectPath = SolutionDirectory.Path;
+            var projectFilePath = GetProjectFilePath(projectPath, languageName);
+
+            CreateNewProject(templateName, projectPath, languageName, TestOutputHelper);
+
+            await AssertProjectLoadsCleanlyAsync(projectFilePath, ignoredDiagnostics ?? []);
+
+            return;
+
+            static string GetProjectFilePath(string projectPath, string languageName)
+            {
+                var projectName = new DirectoryInfo(projectPath).Name;
+                var projectExtension = languageName switch
+                {
+                    LanguageNames.CSharp => "csproj",
+                    LanguageNames.VisualBasic => "vbproj",
+                    _ => throw new ArgumentOutOfRangeException(nameof(languageName), actualValue: languageName, message: "Only C# and VB.NET projects are supported.")
+                };
+                return Path.Combine(projectPath, $"{projectName}.{projectExtension}");
+            }
+
+            static void CreateNewProject(string templateName, string outputPath, string languageName, ITestOutputHelper output)
+            {
+                var language = languageName switch
+                {
+                    LanguageNames.CSharp => "C#",
+                    LanguageNames.VisualBasic => "VB",
+                    _ => throw new ArgumentOutOfRangeException(nameof(languageName), actualValue: languageName, message: "Only C# and VB.NET projects are supported.")
+                };
+
+                var newResult = RunDotNet($"new \"{templateName}\" -o \"{outputPath}\" --language \"{language}\"", output);
+
+                if (DotNetSdkLocator.SdkVersion is not null)
```

Comment:
Rather than having all the logic to discover the SdkVersion, is it easier at this point to just walk up, find a global.json and copy it to the output directory? Then we'll just use whatever is inside of it.

Reply:
Sure. Since the DotNetSdkInstalled condition already locates the repo root for getting the Roslyn solution path, I will do something similar to locate the global.json.